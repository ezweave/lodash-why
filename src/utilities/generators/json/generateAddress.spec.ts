import { 
  generateAddress, 
  generateAllNamedStreetNamePermutations,
  getRandomCity, 
  getRandomGeneratedStreetName, 
  getRandomStreetAddress, 
  getRandomStreetType, 
  StreetType,
} from './generateAddress';

describe('getRandomCity', () => {
  it('gets two random cities', () => {
    const city1 = getRandomCity();
    const city2 = getRandomCity();
    expect(
      city1
    ).not.toEqual(city2);
  });
});

describe('getRandomStreetAddress', () => {
  it('gets a random street addres', () => {
    const streetAddress = getRandomStreetAddress();
    const inRange = streetAddress >= 1 && streetAddress <= 9999;
    expect(inRange).toEqual(true);
  });
});

describe('getRandomStreetType', () => {
  it('gets some random street type', () => {
    const streetType = getRandomStreetType();
    expect(
      StreetType.find(name => name === streetType)
    ).toBeTruthy();
  });
});

describe('generateAllNamedStreetNamePermutations', () => {
  it('generates the same list of random street names', () => {
    expect(
      generateAllNamedStreetNamePermutations()
    ).toMatchSnapshot();
  });
});

describe('getRandomGeneratedStreetName', () => {
  it('generates a random street name', () => {
    const streetName = getRandomGeneratedStreetName();
    expect(streetName).toBeTruthy();
  });
});

describe('generateAddress', () => {
  it('generates a realistic random address', () => {
    const address = generateAddress();
    expect(typeof address === 'object').toEqual(true);
  });
});
