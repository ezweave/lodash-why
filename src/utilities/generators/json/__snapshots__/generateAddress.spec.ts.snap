// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateAllNamedStreetNamePermutations generates the same list of random street names 1`] = `
Array [
  "Boulder Ave",
  "Boulder Blvd",
  "Boulder Ln",
  "Boulder Rd",
  "Boulder St",
  "Broadway Ave",
  "Broadway Blvd",
  "Broadway Ln",
  "Broadway Rd",
  "Broadway St",
  "Elder Ave",
  "Elder Blvd",
  "Elder Ln",
  "Elder Rd",
  "Elder St",
  "Kennedy Ave",
  "Kennedy Blvd",
  "Kennedy Ln",
  "Kennedy Rd",
  "Kennedy St",
  "Lincoln Ave",
  "Lincoln Blvd",
  "Lincoln Ln",
  "Lincoln Rd",
  "Lincoln St",
  "Main Ave",
  "Main Blvd",
  "Main Ln",
  "Main Rd",
  "Main St",
  "Martin Luther King Ave",
  "Martin Luther King Blvd",
  "Martin Luther King Ln",
  "Martin Luther King Rd",
  "Martin Luther King St",
  "Oak Ave",
  "Oak Blvd",
  "Oak Ln",
  "Oak Rd",
  "Oak St",
  "Peachtree Ave",
  "Peachtree Blvd",
  "Peachtree Ln",
  "Peachtree Rd",
  "Peachtree St",
  "Smith Ave",
  "Smith Blvd",
  "Smith Ln",
  "Smith Rd",
  "Smith St",
]
`;
